### PR TITLE
feat(mbb,wbb): prediction & tournament stack — ratings, pregame, in-game WP, resume, bracketology, Monte Carlo

### PR DESCRIPTION
**Pull request by saiemgilani**
## Summary

Adds the full MBB/WBB prediction & tournament stack: opponent-adjusted team ratings, closed-form pregame margin/win-prob/total, an in-game win-probability model, SoS/Quad/WAB résumé, bracketology, and season/bracket Monte Carlo — every model oracle-gated against a committed 2024 corpus (barttorvik, ESPN BPI/predictor/closing lines, and the actual NCAA tournament seeds), with complete women's parity via thin shims over the league-agnostic cores.

## Changes

- **Phase 1 — ratings**: `mbb_team_ratings` (AdjO/AdjD/AdjEM/AdjTempo KenPom-style fixed points). Gate: Spearman vs barttorvik 2024 **0.990**, MAE 2.37.
- **Phase 2 — pregame**: `predict_margin` / `win_prob_from_margin` / `predict_total` + vectorized `mbb_predict_games` (join-key dtype guard). Constants (`em_scale`, HFA, sigma, tempo anchor) fitted by joint least squares on a leakage-free as-of-date walk of 2024 (never rates a game with same-day-or-later data), iterated to a fixed point because the ratings engine reads HFA/tempo. Gates: Brier **0.2006 vs ESPN BPI 0.2031**, spread MAE **1.95** / total MAE **2.90** vs closing lines.
- **Phase 3 — in-game WP**: `in_game_features` + `mbb_in_game_win_prob`, scored from a bundled shallow-xgboost artifact (`mbb/models/mbb_in_game_wp.ubj`, trained on 2023 pbp; xgboost is a lazy optional import). A plain logistic failed the decile gate (max gap 0.133 — no time×score×pregame interactions), so this uses the plan's sanctioned tree escalation. Gate: out-of-sample 2024 max decile gap **0.0298** (≤ 0.03).
- **Phase 4 — résumé**: `mbb_strength_of_schedule` (SoS, NET-style Quad 1–4 records with venue-adjusted thresholds, quality wins, Wins Above Bubble reusing the Phase-2 predictors). Gate: SoS Spearman vs ESPN BPI SOS **0.923**.
- **Phase 5 — bracketology**: `mbb_bracketology` / `project_bracket` + `_conference_auto_bids` (résumé blend → 68-team field → seeds, at-large logistic centred on the selection cutoff). Gate: seed-order Spearman vs the **actual 2024 committee** **0.938** (oracle swapped from the bracketmatrix consensus — its archive is unreachable behind a self-signed cert, and the real committee list is the stronger reference; seeds captured from ESPN scoreboard `curatedRank`).
- **Phase 6 — Monte Carlo**: `simulate_game` / `mbb_season_sim` / `mbb_bracket_sim` (seeded `default_rng`, deterministic). Gate: win-prob calibration slope **1.03** on all 413 neutral-site 2024 games (the 67 tournament games alone have bootstrap slope SE 0.31 — statistically unresolvable against a ±0.1 band, so the gate uses the powered neutral-site population that contains them).
- **Phase 7 — WBB parity**: women's constants fitted on the WBB 2024 backtest, a women's-trained WP artifact, six thin shims (`wbb_team_ratings`, `wbb_predict_games`, `wbb_in_game_win_prob`, `wbb_strength_of_schedule`, `wbb_bracketology`, `wbb_season_sim`/`wbb_bracket_sim`) over the mbb cores with league-aware loader dispatch, and every gate re-run on WBB 2024 at the same thresholds: Torvik **0.995**, SoS **0.985**, seed-order **0.976**, WP deciles **0.0224**, neutral slope 0.917 (the two observed-floor gates carry women's documented floors: spread 3.61→4.0, total 5.56→6.5 — WBB book coverage is thin, 97 games with a usable close).
- **Fixtures**: committed 2024 oracle corpora under `tests/fixtures/{mbb,wbb}_prediction/` (16 parquets + provenance READMEs) so all gates run offline in CI.
- **Fix folded in**: the Phase-0 torvik fixture had 17 State schools keyed to their flagship's ESPN `team_id` (set-ordered name-match let the lossy bare-base candidate beat the "X State" expansion); the ordered matcher moved the Phase-1 gate from 0.952 → 0.990.

## Type of change

- [x] New feature (non-breaking)

## Test plan

- [x] `uv run pytest tests/mbb tests/wbb` passes offline (679 passed — all oracle/backtest gates run offline against the committed fixtures)
- [x] `uv run pytest tests/mbb tests/wbb tests/nba` passes (1063 passed)
- [x] `uv run mypy` passes (119 files; all 12 new modules on the ratchet)
- [x] `uv run ruff check sportsdataverse/ tests/` passes
- [x] `uv run python tools/codegen/generate.py --check` clean (reference docs regenerated)
- [x] Fitting/training/capture scripts re-runnable from `dev/mbb_prediction/` (`fit_pregame.py`, `train_in_game_wp.py`, `capture_oracle.py`; `PRED_LEAGUE=womens` for WBB); fits verified idempotent at the committed constants
- [x] `sdv-toolkit:polars-1x-reviewer` (0 findings across 18 files) + `sdv-toolkit:returns-table-auditor` lenses run; audit findings fixed in the final commit

## Breaking changes

None.

## Documentation

- [x] CHANGELOG.md entry added
- [x] Docstring(s) updated (Google-style + Example blocks on every public callable; generated reference pages under `docs/docs/{mbb,wbb}/reference/additional.md`)
- [ ] CLAUDE.md / copilot-instructions.md updated (no convention change)

## Checklist

- [x] My code follows the project's [code standards](../CONTRIBUTING.md#code-standards-for-new-modules).
- [x] I have NOT included AI agents (Claude, Copilot, GPT, etc.) as commit co-authors.
- [x] I have searched existing PRs to confirm this isn't a duplicate.

## Summary by CodeRabbit

* **New Features**
  * Added men’s and women’s prediction & tournament stack: win probability, in-game win probability, team ratings, strength-of-schedule, bracketology, and season/bracket Monte Carlo simulation.
  * Added shared league constants via `LeagueConstants`/`get_constants`, plus additional prediction and evaluation helpers.
* **Documentation**
  * Expanded MBB/WBB reference docs and updated Unreleased changelog entries/TOCs, including offline oracle fixture notes.
* **Tests**
  * Added unit tests and oracle-based backtests for predictions, simulations, bracketology, calibration/metrics, and women’s parity shims.

### CODE REVIEW

File: sportsdataverse/mbb/mbb_game_predict.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+            feats = in_game_features(pbp, 0.62)
+    """
+    logit = math.log(pregame_home_prob / (1.0 - pregame_home_prob))
+    return pbp.select(
+        (pl.col("home_score") - pl.col("away_score")).cast(pl.Float64).alias("score_diff"),
+        pl.col("start_game_seconds_remaining").cast(pl.Float64).clip(lower_bound=0.0).alias("sec_left"),
+        pl.col("start_game_seconds_remaining").cast(pl.Float64).clip(lower_bound=0.0).sqrt().alias("sqrt_sec_left"),
+        pl.lit(logit, dtype=pl.Float64).alias("pregame_logit"),
+        (pl.col("team_id") == pl.col("home_team_id")).fill_null(False).cast(pl.Int8).alias("home_has_ball"),
+    )
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/mbb/mbb_game_predict.py:90]: Fixed in eea36379 — `pregame_home_prob` is clipped to `[1e-6, 1 - 1e-6]` before the logit (with a boundary regression test for exact 0.0/1.0 inputs).

File: sportsdataverse/mbb/mbb_strength_of_schedule.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        mbb_team_ratings,
+    )
+
+    load_schedule, _ = _league_loaders(league)
+    results = _normalize_schedule(load_schedule(seasons)).filter(
+        pl.col("home_score").is_not_null() & pl.col("away_score").is_not_null()
+    )
+    ratings = mbb_team_ratings(seasons, league=league)
+    out = strength_of_schedule(results, ratings, league=league)
+    return out.to_pandas() if return_as_pandas else out
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/mbb/mbb_strength_of_schedule.py:245]: Fixed in eea36379 — `mbb_strength_of_schedule` now loads the schedule/boxscore once and builds the ratings inline (`raw_game_efficiency` + `adjust_efficiency` + dense rank, matching what `mbb_team_ratings` produced), same pattern as `mbb_bracketology`.
